core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/CombinedGroupsRule.scala

```diff
@@ -17,19 +17,14 @@
 package tech.beshu.ror.accesscontrol.blocks.rules.auth
 
 import tech.beshu.ror.accesscontrol.blocks.rules.Rule
-import tech.beshu.ror.accesscontrol.blocks.rules.Rule.{Name, RuleName}
-import tech.beshu.ror.accesscontrol.domain.{CaseSensitivity, GroupsLogic}
+import tech.beshu.ror.accesscontrol.blocks.rules.auth.base.BaseGroupsRule
+import tech.beshu.ror.accesscontrol.blocks.rules.auth.base.BaseGroupsRule.{Creator, Settings}
+import tech.beshu.ror.accesscontrol.domain.*
 
-final class GroupsAndRule(override val settings: BaseGroupsRule.Settings[GroupsLogic.And],
-                          override val userIdCaseSensitivity: CaseSensitivity)
-  extends BaseGroupsRule(settings) {
+class CombinedGroupsRule(override val name: Rule.Name, override val settings: Settings[GroupsLogic.Combined])
+                        (override implicit val userIdCaseSensitivity: CaseSensitivity)
+  extends BaseGroupsRule[GroupsLogic.Combined](name, settings)
```

Comment:
this rule should also have a fixed name. I suggest `groups_combined` event if this name won't be used as a yaml key. But it will be used in audit and logs. 

Alternative proposition would be `groups_{all_of/any_of}_{not_all_of/not_any_of}`. 

WDYT?

Reply:
Done, with `groups_combined`. Using the second proposition would require refactor and doing some workaround for type erasure.